src/typings.ts

```diff
@@ -420,17 +420,35 @@ export interface Signature {
   concatSig: string
 }
 
-export interface IdentityWalletSchema {
-  TLWalletVersion: number
-  ethersKeystore: string
-  identityAddress: string
-  walletType: string
+// WALLET
+export type WalletTypeEthers = 'ethers'
+export type WalletTypeIdentity = 'identity'
+export type WalletType = WalletTypeEthers | WalletTypeIdentity
+
+export interface TLWalletSchema {
+  version: number
+  type: WalletType
+  meta?: any
+}
+
+export interface SigningKey {
+  privateKey: string
+  mnemonic: string
+  derivationPath: string
+}
+
+interface EthersWalletMeta {
+  signingKey: SigningKey
+}
+
+export interface EthersWalletSchema extends TLWalletSchema {
+  type: WalletTypeEthers
+  meta: EthersWalletMeta
 }
 
-export interface EthersWalletSchema {
-  TLWalletVersion: number
-  ethersKeystore: string
-  walletType: string
+export interface IdentityWalletSchema extends TLWalletSchema {
+  type: WalletTypeIdentity
+  meta: EthersWalletMeta
```

Comment:
I would here also store the address, so that you dont have to recalculate the address.